src/browser/browser.js

```diff
@@ -168,6 +169,32 @@ define((require, exports, module) => {
     position: 'relative'
   };
 
+
+  // History API hooks
+  const history = new History({withTop: true});
+
+  const beginVisit = ({webView, time}) => {
+    history.edit(Site.from(webView), Site.beginVisit({id: webView.id, time}));
+  };
+
+  const endVisit = ({webView, time}) => {
+    history.edit(Site.from(webView), Site.endVisit({id: webView.id, time}));
+  };
+
+  const changeTitle = ({webView, title}) => {
+    history.edit(Site.from(webView), site => site.set('title', title));
+  };
+
+  const changeImage = ({webView, image}) => {
+    history.edit(Site.from(webView), site => site.set('image', image));
+  };
+
+  const changeIcon = ({webView, icon}) => {
```

Comment:
Should these 3 things be a single function? That way we could batch changes if we were changing more than one thing at once.